docs/specs/code/COIN_PRICE.md

```diff
@@ -0,0 +1,161 @@
+# Abstract
+
+Describe the ability to see and input ADA amount in another currency.
+
+# Proposal
+
+This feature includes two aspects:
+
+## 1. Input
+When the UI contains an ADA amount input, it should let the user input the number in another currency and covert it to ADA amount at the present conversion rate.
+
+## 2. Output
+When the UI shows an ADA amount, additionally it should be able to show the amount in another currency. If the ADA amount is related to an existing transaction, the user will be able to see the price at two different moments:
+-right now
+-at the moment of the transaction.
+
+The following fiat currencies will be supported:
+
+- USD	US dollar	
+- JPY	Japanese yen
+- BGN	Bulgarian lev
+- CZK	Czech koruna
+- DKK	Danish krone
+- GBP	Pound sterling
+- HUF	Hungarian forint
+- PLN	Polish zloty	
+- RON	Romanian leu	
+- SEK	Swedish krona	
+- CHF	Swiss franc	
+- ISK	Icelandic krona
+- NOK	Norwegian krone
+- HRK	Croatian kuna
+- RUB	Russian rouble
+- TRY	Turkish lira
+- AUD	Australian dollar
+- BRL	Brazilian real	
+- CAD	Canadian dollar
+- CNY	Chinese yuan renminbi
+- HKD	Hong Kong dollar
+- IDR	Indonesian rupiah
+- ILS	Israeli shekel	
+- INR	Indian rupee	
+- KRW	South Korean won
+- MXN	Mexican peso	
+- MYR	Malaysian ringgit
+- NZD	New Zealand dollar
+- PHP	Philippine peso	
+- SGD	Singapore dollar
+- THB	Thai baht	
+- ZAR	South African rand
+
+These are the currencies supported by https://exchangeratesapi.io/. See the Architecture section for details.
+
+The following cryptos will be supported:
+- BTC
+- ETH
+
+# Terminology
+
+Frontend: yoroi-frontend.
+
+Backend: yoroi-backend-service, partically the extensions that are required to implement this proposal.
+
+API: third-party services that provide the price data.
+
+Price data: the conversion rates between ADA and all supported currencies.
+
+# Architecture
+
+The frontend refreshes the current ADA price at 15 minute interval by querying the backend.
+
+The backend refreshes the current ADA price at 15 minute interval by querying the API. So the freshness of the price data from the user's perspective is 15 minutes + 15 minutes = 30 minutes. The backend stores the timestamped price data in a historical price database.
```

Comment:
This is way too high. 30 minute is enough for the price to change by multiple percentage points.

Reply:
What is the appropriate number or method to determine this number?

Reply:
I feel that every 1-5 min seems like the right number from the user perspective

Reply:
Thought: maybe to simplify we should just query the price at the start of every slot. Makes it easier to associate a price  to a transaction.